server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -285,6 +285,7 @@ private void performPhaseOnShard(final int shardIndex, final SearchShardIterator
             Runnable r = () -> {
                 final Thread thread = Thread.currentThread();
                 try {
+                    final SearchPhase phase = this;
```

Comment:
Do you need this?

Reply:
yes, onPhaseFailure() below need the SearchPhase.